File: 01_09_rotate_string.cpp
/** Solution 1 **/
class Solution {
public:
    bool rotateString(string s, string goal) {

        if (s.size() == goal.size())
        {
            for(auto i = 0; i < s.length(); i++)
            {
                s = s.at(s.length() - 1) + s.substr(0, s.length() - 1);

                if(s == goal) 
                  return true;
            }
        }
        return false;
    }
};

/** Solution 2 **/
class Solution {
public:
    bool rotateString(string s, string goal) {

        if (s.size() != goal.size())
            return false;

        s += s;

        return s.find(goal) != -1;
    }
};

File: 01_09_rotate_string.js
/**
 * @param {string} s
 * @param {string} goal
 * @return {boolean}
 */
 var rotateString = function(s, goal) {
     let newS = s.split('')
     //['a', 'b', 'c', 'd', 'e']
     let count = 0
     let arrayLengthCount = 0 
      while(arrayLengthCount < newS.length) {
        //   console.log(newS)
          //['a', 'b', 'c', 'd', 'e']
        newS[newS.length] = newS[count]
        //['a', 'b', 'c', 'd', 'e', 'a']
        newS.splice(count, 1)
        let newString = newS.join('')
        // console.log(newS)
        if(newString === goal) {
            console.log(newString)
            return true;
        }
        arrayLengthCount++
    }
    return false;
};


